test/re/frida/MethodTest.java

```diff
@@ -55,14 +59,183 @@ public void replacementPropagatesExceptions() {
         thrown.expectMessage("Already dead");
         badger.die();
     }
+    
+    @Test
+    public void interfaceCannotBeInstantiated() {
+        loadScript("var X509TrustManager = Java.use('javax.net.ssl.X509TrustManager');" +
+                "try {" +
+                "  var tm = X509TrustManager.$new();" +
+                "  send('ok');" +
+                "} catch (e) {" + 
+                "  send('couldnt create trustmanager');" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }
+    
+    @Test
+    public void genericReturnJavaLangClass() {
+        loadScript("var c = Java.use ('java.lang.Class');" +
+                "try {" +
+                "  var orig = c.forName.overload('java.lang.String');" +
+                "  c.forName.overload('java.lang.String').implementation = function (s) {" +
+                "    orig.call(this,s);" +
+                "  };" +
+                "  var d = c.forName('re.frida.MethodTest');" +
+                "  send('ok');" +
+                "} catch (e) {" + 
+                "  send('class.forName failed. ' + e);" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }
+    
+    @Test
+    public void genericReturnBadger() {
+        loadScript("var c = Java.use('re.frida.Badger');" +
+                "try {" +
+                "  var orig = c.forName;" +
+                "  c.forName.implementation = function () { " +
+                "    orig.call(this);" +
+                "  };" +
+                "  var d = c.forName();" +
+                "  send('ok');" +
+                "} catch (e) {" + 
+                "  send('forName failed. ' + e);" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }
+    
+    // this one still just producing 
+    // Error: access violation accessing 0xf2b295fe
+    @Test
+    public void nativeReturnGenericVmStack() { 
+        loadScript(
+                "try {" +
+                "  var c = Java.use('dalvik.system.VMStack');" +
+                "  var orig = c.getStackClass2;" +
+                "  c.getStackClass2.implementation = function () {" +
+                "    orig.call(this);" +
+                "  };" +
+                "  var stack = c.getStackClass2();" +
+                "  send('ok');" +
+                "} catch(e) {" + 
+                "  send('nativeReturnGeneric: ' + e);" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }
+    
+    // this one still just producing 
+    // Error: access violation accessing 0x2133c66a
+    @Test
+    public void nativeReturnGenericBadgerWrapperAroundJavaLangClass() { 
+        loadScript(
+                "try {" +
+                "  var c = Java.use('re.frida.Badger');" +
+                "  var orig = c.forNameYo;" +
+                "  c.forNameYo.implementation = function () {" +
```

Comment:
This reads out the same value twice making the code hard to read. (See comment below.)